### DISCUSSION TITLE
Unable to upload image with Cross-Origin Read Blocking warning

### DISCUSSION THREAD

**Discussion by dev-nicelee** [outline/outline#1397] [General]
It is unable to upload image with Cross-Origin Read Blocking warning

For your better understanding of the situation, more information as below:
- currently reverse proxy is set by nginx for https of self-hosted outline.
- The url is [https://wiki.solidware.io](https://wiki.solidware.io). 
- For uploading a image, dockerlized minio is up in the same server.
- the minio server is also reverse proxied by nignx and url is [https://upload-wiki.solidware.io](https://upload-wiki.solidware.io).
- Uploading is successful if I check minio server, but fetching image is problematic.

**Warning Message in the browser console**
```
Cross-Origin Read Blocking (CORB) blocked cross-origin response https://upload-wiki.solidware.io/minio/solidware-wiki-uploads/uploads/9ab3d086-b957-49da-825d-4705fdf1c500/8fbd65a9-ef09-45cf-82f6-066b6e4a106f/1_TnGnN4mSbfcvVczYYh2m7g.png with MIME type text/html. See https://www.chromestatus.com/feature/5629709824032768 for more details.
```

![Screen Shot 2020-07-28 at 8 01 51 PM](https://user-images.githubusercontent.com/32350596/88657684-46c02900-d10d-11ea-9dbf-60455d02b5b4.png)

If it is more related to minio server settings, please let me know about it.

**Comment by ReneGraetz** [comment]
Did you ever find a solution for that? I have the exact Problem.

**Comment by EacAyf** [comment]
I had this same issue, by adding the header below into the 'location' bracket of your Nginx you should get a little further. 

`add_header 'Access-Control-Allow-Origin' 'https://minio.example.com/bucket';`

This link might help (although I don't want you to use the wildcard they show)
https://enable-cors.org/server_nginx.html

**Reply by almereyda** [reply]
Did this suggestion work for you @dev-nicelee, @ReneGraetz and @wenqvip, and can be marked as an accepted answer?

**Reply by dev-nicelee** [reply]
I'm sorry I haven't tried the suggestion above. I switched to AWS S3 bucket after struggling some.. hope someone tried this and worked!🙏🏻

**Reply by laserg** [reply]
Colleagues, I found the real reason
**Research:**
First of all CORB not eq CORS
The reason of CORB related issue is MIME type of response is not expected.
The client expects image but gets text/html.
To see it clearly run chrome with disabled CORS/CORB ("C:\Program Files\Google\Chrome\Application\chrome.exe" --disable-web-security --user-data-dir="C:/ChromeDevSession")

**Solution**
1. Disable minio UI-CONSOLE via use env vareable MINIO_BROWSER=off
2. Expose port 9000 and map it via proxy if you need
3. Have fun !!!

My configuration with traefik proxy for ex.
* Minio container
![image](https://user-images.githubusercontent.com/6731215/143147080-5c49009a-369c-42a7-a867-e85aa857ba20.png)
* Outline container
![image](https://user-images.githubusercontent.com/6731215/143147557-25901a00-838d-4b27-ad77-622c36f4bfa6.png)
